backend/manager/modules/bll/src/main/java/org/ovirt/engine/core/bll/storage/lsm/LiveMigrateDiskCommand.java

```diff
@@ -214,9 +220,29 @@ protected CreateSnapshotForVmParameters getCreateSnapshotParameters() {
         params.setDiskIdsToIgnoreInChecks(getImageGroupIds());
         params.setNeedsLocking(false);
         params.setEndProcedure(EndProcedure.COMMAND_MANAGED);
+
+        updateImageInitialSizeInBytes(params);
+
         return params;
     }
 
+    private void updateImageInitialSizeInBytes(CreateSnapshotForVmParameters params) {
+        if (FeatureSupported.isLiveStorageMigrationInitialSizeSupported(getVm())) {
+            StorageDomainStatic sourceDomain = storageDomainStaticDao.get(getParameters().getSourceStorageDomainId());
+            if (sourceDomain.getStorageType().isBlockDomain()) {
+                int volumeUtilizationChunkInMB = Config.getValue(ConfigValues.VolumeUtilizationChunkInMB);
+                long imageInitialSizeInBytes = 3 * SizeConverter.convert(volumeUtilizationChunkInMB,
+                        SizeConverter.SizeUnit.MiB,
+                        SizeConverter.SizeUnit.BYTES).longValue();
+                // TODO: do I need to create a fresh copy of the original disk here?
+                //DiskImage disk = DiskImage.copyOf(getDiskImage());
+                DiskImage disk = getDiskImage();
+                disk.setInitialSizeInBytes(Math.min(imageInitialSizeInBytes, disk.getSize()));
```

Comment:
We need to create new disk image, and set the initial size:

   DiskImage diskParam = new DiskImage()
   diskParams.setInitialSizeInBytes(Math.min(imageInitialSizeInBytes, disk.getSize()));
   params.setDiskImagesMap(Collections.singletonMap(disk.getId(), diskParams));

Reply:
I think getSize() is the virtual size (capacity in vdsm), and getActualSizeInBytes is the actual size (apparentsize in vdsm), but this is only a guest. There code should document this clearly.